torchinductor/codegen/common.py

```diff
@@ -82,6 +85,15 @@ def prune(index):
     return [x for x in sizes if x is not None], reindex, prune
 
 
+def index_prevent_reordering(index: typing.List[sympy.Expr], index_vars, sizes):
+    from ..ir import FlexibleLayout
+
+    res = index
+    # added contiguous index prevents reordering
+    res.append(sympy_dot(index_vars, FlexibleLayout.contiguous_strides(sizes)))
```

Comment:
```suggestion
    return [*index, sympy_dot(index_vars, FlexibleLayout.contiguous_strides(sizes))]
```

Mutating inputs to a function can lead to confusing bugs.